pkg/registry/entry_test.go

```diff
@@ -0,0 +1,341 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEntryContains(t *testing.T) {
+	var tests = []struct {
+		first, second Entry
+		res           bool
+	}{
+		{
+			first:  ParseEntry("192.168.0.1"),
+			second: ParseEntry("192.168.0.1"),
+			res:    true,
+		},
+		{
+			first:  ParseEntry("192.168.0.1"),
+			second: ParseEntry("192.168.0.1/16"),
+			res:    false,
+		},
+		{
+			first:  ParseEntry("192.168.0.1"),
+			second: ParseEntry("192.168.0.2"),
+			res:    false,
+		},
+		{
+			first:  ParseEntry("192.168.0.1/24"),
+			second: ParseEntry("192.168.0.11"),
+			res:    true,
+		},
+		{
+			first:  ParseEntry("172.16.0.0/12"),
+			second: ParseEntry("172.17.0.0/24"),
+			res:    true,
+		},
+		{
+			first:  ParseEntry("172.16.0.0/12"),
+			second: ParseEntry("172.15.0.0/24"),
+			res:    false,
+		},
+		{
+			first:  ParseEntry("172.16.0.0/12"),
+			second: ParseEntry("*.google.com"),
+			res:    false,
+		},
```

Comment:
worth adding an IP and hostname pair to make the match or lack explicit between FQDN and IP